src/components/calcite-dropdown/calcite-dropdown.e2e.ts

```diff
@@ -753,6 +753,60 @@ describe("calcite-dropdown", () => {
     expect(await dropdownWrapper.isVisible()).toBe(false);
   });
 
+  describe("opens the dropdown with click, enter, or space", () => {
+    it("opens when dropdown-trigger is a button", async () => {
+      const page = await newE2EPage();
+      await page.setContent(html` 
+        <calcite-dropdown>
+          <calcite-button slot="dropdown-trigger">Open dropdown</calcite-button>
+          <calcite-dropdown-group selection-mode="single">
+            <calcite-dropdown-item id="item-1">
+              Dropdown Item Content
+            </calcite-dropdown-item>
+            <calcite-dropdown-item id="item-2" active>
+              Dropdown Item Content
+            </calcite-dropdown-item>
+          </calcite-dropdown-group>
+          </calcite-dropdown>
+        </calcite-dropdown> 
+      `);
+      const element = await page.find("calcite-dropdown");
+      const trigger = await element.find("calcite-button[slot='dropdown-trigger']");
+      const dropdownWrapper = await page.find("calcite-dropdown >>> .calcite-dropdown-wrapper");
+
+      expect(await dropdownWrapper.isVisible()).toBe(false);
+      await trigger.click();
+      await page.waitForChanges();
+      expect(await dropdownWrapper.isVisible()).toBe(true);
```

Comment:
Should we also test a space/enter key here? Maybe close it and try again with space and enter key. The test name can be a bit deceiving.

Reply:
Maybe also test that the event is only called once each time?

Reply:
Added space/enter to tests + check for how many times the event has been called. Adam pointed out that trigger slot should support any slotted content, eg link, div, calcite-link, etc. Removing call for click handler on keyboard interaction removed that ability, so I added it back. Instead, I added a check for target.nodeName !== "CALCITE-ACTION” to exclude that along with the button. Matt was suggesting to get rid of checking node types, as these very feeble and can easily break if the nodeNames change. But that this can be handled in another pr. I added some demos so I can see it in action, let me know if I should remove these.